### PR TITLE
feat: Reimbursement posts in slack along with donation tracking

### PR DESCRIPTION
**Pull request by mjmaurer**
This feature relies on a new base (P2PMoney) with the tables Donors, PaymentRequests, and DonorPayments. Demos/flows can be found [in this doc](https://docs.google.com/document/d/19dWzBmBnuQYVmsbczSCIo2dcrO9zJHKNyg-wy8sbPsY/edit).

After this feature:
- When a new PaymentRequest is created, a post is created in community_reimbursements
- As people comment, a new DonorPayment row is created for each payment
- Donor rows are external donors. When a new DonorPayment is created for an external donor, it will be posted to the relevant PaymentRequest thread

### CODE REVIEW

File: src/workers/airtable-sync/actions/payments/newPaymentRequest.js [hunk 1]
```diff
@@ -1,15 +1,176 @@
...
+  
+${str("slackapp:reimbursementBotPost.post.message.outro")}`;
+}
+
+const handleExistingPaymentRequest = async (
+  existingPaymentRequest,
+  code,
+  reimbursementChannel
+) => {
+  console.log(`Handling duplicate paymentRequest for code: ${code}`);
```
Comment (alexquick) [inline src/workers/airtable-sync/actions/payments/newPaymentRequest.js:150]: niiiice

File: src/workers/airtable-sync/actions/payments/updateReimbursementStatus.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+const slackapi = require("~slack/webApi");
```
Comment (alexquick) [inline src/workers/airtable-sync/actions/payments/updateReimbursementStatus.js:1]: I feel like not worrying about DRYing up the duplication here w/r/t the other message that gets updated by the bot was a good call.

File: src/workers/airtable-sync/actions/payments/updateReimbursementStatus.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      ":heavy_check_mark: REIMBURSED\n"
+    );
+  }
+  return str(
+    "slackapp:reimbursementBotPost.post.statusPrefix.default",
+    ":red_circle:"
+  );
+}
+
+async function getExistingMessage(ts, channel) {
```
Comment (alexquick) [inline src/workers/airtable-sync/actions/payments/updateReimbursementStatus.js:59]: though this looks familiar, if it got moved to `~lib/slack/??` would it be useful elsewhere?

File: index.js [hunk 4]
```diff
@@ -134,7 +134,7 @@ const airtableIntervalMs = parseInt(process.env.AIRTABLE_SYNC || 0);
 if (airtableIntervalMs > 0) {
   airtableWorker(airtableIntervalMs);
   if (process.env.AIRTABLE_PAYMENTS_BASE) {
-    airtablePaymentsWorker(airtableIntervalMs);
+    airtablePaymentsWorker(airtableIntervalMs + 2000); // Stagger polling to avoid rate limit
```
Comment (alexquick) [inline index.js:137]: I was thinking if this becomes a serious problem we can make something to automatically enforce the rate limit. We could make a facade around the instance that comes out of `airtable("tableName")` that pauses before `select`/`update`/ etc if needed.

File: src/lib/airtable/tables/paymentRequests.js [hunk 5]
```diff
@@ -11,7 +11,7 @@ exports.updatePaymentRequestByCode = async (code, update) => {
   try {
     const records = await paymentRequestsTable
       .select({
-        filterByFormula: `({${fields.requestCode}} = '${code}')`
+        filterByFormula: `(FIND('${code}', {${fields.requestCode}}) > 0)`
```
[the remark this answers is not in the document]
Comment (mjmaurer) [reply src/lib/airtable/tables/paymentRequests.js:14]: Good point. The whole idea of 4-6 letters was only a hotfix. We could go back to 6 letters when 4 letter slack posts have all been completed.

File: src/lib/airtable/tables/paymentRequests.js [hunk 6]
```diff
@@ -36,6 +36,50 @@ exports.updatePaymentRequestByCode = async (code, update) => {
   }
 };
 
+exports.findPaymentRequestsInSlack = async code => {
```
[the remark this answers is not in the document]
Comment (mjmaurer) [reply src/lib/airtable/tables/paymentRequests.js:39]: good point

File: src/workers/airtable-sync/actions/payments/updateReimbursementStatus.js (hunk 2 above)
Comment (mjmaurer) [reply src/workers/airtable-sync/actions/payments/updateReimbursementStatus.js:1]: Yea it was something I considered. Maybe in the future once this stabalizes a bit!

File: index.js (hunk 4 above)
Comment (mjmaurer) [reply index.js:137]: Definitely a good idea

**Comment by mjmaurer** [comment]
@alexquick  Thanks for reviewing. Part of the reason I pushed against automated texts to potential donors was so I could sweat a little less about race conditions :/

**Pull request merged**